### PR TITLE
Added support for Win32 build

### PR DESCRIPTION
**Pull request by fredzo**
Hi Gabriel,

If you're willing to take this PR, I added some documentation and made some changes to support compiling on an windows machine and building and running the hybrid application on a win32 platform.

Have a very nice new year's eve !

Frederic.

### CODE REVIEW

File: system/os_host/library/spf/spf.c [hunk 1]
```diff
@@ -160,6 +160,7 @@ int sfp_printi(char **out, int i, int b, int sg, int width, int pad, int letbase
 	return pc + sfp_prints (out, s, width, pad);
 }
 
+#ifndef WIN32
```
Comment (gabonator) [inline system/os_host/library/spf/spf.c]: Could you please revert those additions and exclude spf completely from windows build?

os_platform/common/source/framework/framework.cpp:
```c
#if !defined(__APPLE__) && !defined(WIN32)
#include "../../../os_host/library/spf/spf.h"
#include "../../../os_host/library/spf/spf.c"
#endif
```

File: system/os_platform/win32_sdl/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,132 @@
+cmake_minimum_required(VERSION 2.8)
```
Comment (gabonator) [inline system/os_platform/win32_sdl/CMakeLists.txt]: do you use this file? I wanted to remove this one (mac_sdl/CMakeLists.txt and mac_sdl/build.sh), it was just for testing. All projects/applications should have their own cmakelists file in their directory. Currently it is a little bit messy, but in future I would like to have single cmake file to build the application for all platforms (right now there are shell scripts for arm builds)

File: system/os_host/library/spf/spf.c (hunk 1 above)
Comment (fredzo) [reply system/os_host/library/spf/spf.c]: Done !

File: system/os_platform/win32_sdl/CMakeLists.txt (hunk 2 above)
Comment (fredzo) [reply system/os_platform/win32_sdl/CMakeLists.txt]: Well I made my tests with this file and then discovered you created dedicated CMakeLists files for each application ;-)
I will have to adapt the CMakeLists files to switch between Win32 and Mac hal according to the platform.
I'll create a separate PR for that if you're OK

Comment (fredzo) [reply system/os_platform/win32_sdl/CMakeLists.txt]: Well I ended up making the change in gabosc specific CMakeList file : added a if(WIN32) for win32 hal files + math.h override.

Comment (fredzo) [reply system/os_platform/win32_sdl/CMakeLists.txt]: I also added elfstrip binary for win32 + a CMakeLists file to generate it's project

**Pull request merged**